test/sqllogictest/dates-times.slt

```diff
@@ -1368,3 +1368,212 @@ query T
 SELECT now() + null;
 ----
 NULL
+
+# Test timestamp precision.
+query error Expected literal unsigned integer
+SELECT '1970-01-01T00:00:00.666666'::timestamp(-1);
```

Comment:
Let's also put a few of these interesting cases in the parser tests, e.g. in `src/sql-parser/tests/testdata/literal` (the prefixed values) and `src/sql-parser/tests/testdata/scalar` (the casts). Doesn't need to be many, just want to make sure they get hit.

Once you write the tests, you can auto-generate the valid values using:

```
REWRITE=1 cargo test --package mz-sql-parser --test sqlparser_common -- datadriven --exact
```

Reply:
done!